imports/plugins/core/versions/server/util/hashProduct.js

```diff
@@ -0,0 +1,121 @@
+import hash from "object-hash";
+import Logger from "@reactioncommerce/logger";
+
+const productFieldsThatNeedPublishing = [
+  "_id",
+  "ancestors",
+  "description",
+  "facebookMsg",
+  "googleplusMsg",
+  "handle",
+  "hashtags",
+  "isDeleted",
+  "isVisible",
+  "media",
+  "metaDescription",
+  "metafields",
+  "originCountry",
+  "pageTitle",
+  "parcel",
+  "pinterestMsg",
+  "productType",
+  "price",
+  "pricing",
+  "publishedScope",
+  "shopId",
+  "supportedFulfillmentTypes",
+  "template",
+  "title",
+  "twitterMsg",
+  "type",
+  "variants",
+  "vendor"
+];
+
+/**
+ *
+ * @method getTopLevelProduct
+ * @summary Get a top level product based on provided ID
+ * @param {string} productOrVariantId - A variant or top level Product Variant ID.
+ * @param {Object} collections - Raw mongo collections.
+ * @return {Promise<Object[]>} Top level product object.
+ */
+async function getTopLevelProduct(productOrVariantId, collections) {
+  const { Products } = collections;
+
+  // Find a product or variant
+  let product = await Products.findOne({
+    _id: productOrVariantId
+  });
+
+  // If the found product has ancestors,
```

Comment:
I searched a bit and didn't find any other util function or any other code exactly like this.

Reply:
I think either this might be it or maybe add this function to here? https://github.com/reactioncommerce/reaction/blob/feat-4725-aldeed-extend-product-attr/lib/api/catalog.js#L192